### PR TITLE
Better collision checking

### PR DESCRIPTION
**Pull request by Creeperface01**
i named this method getCollisionBoundingBox() because idk how should i name it :D so you can rename it or tell me better name

### CODE REVIEW

**Comment by Creeperface01** [comment]
wait

**Comment by MagicDroidX** [comment]
`getBlock` is a time-consuming method so in my opinion, this is undoubtedly slower.

**Comment by Creeperface01** [comment]
its slower if move event is canceled but its not very often

**Comment by Creeperface01** [comment]
or what should we use instead of getBlock()? getBLockIdAt() ?

**Comment by Creeperface01** [comment]
it won't be slower than now

**Comment by MagicDroidX** [comment]
I've made a conclusion here: https://github.com/Nukkit/Nukkit/pull/1128
PTAL

**Comment by Creeperface01** [comment]
OK but with this change you can check blocks without affecting performance too. Because blocksAround or blockUnder lists contains only few blocks

**Comment by Creeperface01** [comment]
but in entity base tick we even get these blocks so i dont know why should be this slow

**Comment by Creeperface01** [comment]
i removed underBlocks, but if someone would want to get them in playerMOveEvent, he can use getBlocksUnder() and in base tick nukkit won't find these blocks again

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -1849,6 +1851,7 @@ public void handleDataPacket(DataPacket packet) {
 
                         this.setRotation(movePlayerPacket.yaw, movePlayerPacket.pitch);
                         this.newPosition = newPos;
+                        this.newPosition = newPos;
```
Comment (Creeperface01) [inline src/main/java/cn/nukkit/Player.java]: WTF :D thx

**Comment by Creeperface01** [comment]
Why?

**Pull request closed without merging**